File: backend/src/vaccination/vaccination.entity.ts
import DoctorEntity from "src/doctor/doctor.entity";
import PatientEntity from "src/patient/patient.entity";
import { Column, Entity, ManyToOne, PrimaryGeneratedColumn } from "typeorm";
import { JsonProperty, Serializable } from "typescript-json-serializer";

@Serializable()
@Entity()
export default class VaccinationEntity {
  @JsonProperty()
  @PrimaryGeneratedColumn("uuid")
  public id: string;

  @JsonProperty()
  @ManyToOne(() => PatientEntity, (patient) => patient.vaccinationsInProgress)
  public patient: PatientEntity;

  @JsonProperty()
  @ManyToOne(() => DoctorEntity, (doctor) => doctor.coordinatedVaccinations)
  public vaccinator: DoctorEntity;

  @JsonProperty()
  @Column()
  public scheduledDate: Date;

  @JsonProperty()
  @Column("simple-array")
  public offeredAppointments: Date[];
}

File: vaccination_helper/lib/core/auth/vaccination/vaccination_entity.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:vaccination_helper/core/doctor/doctor_entity.dart';
import 'package:vaccination_helper/core/patient/patient_entity.dart';

part 'vaccination_entity.g.dart';

@JsonSerializable()
class VaccinationEntity {
  String id;
  PatientEntity patient;
  DoctorEntity vaccinator;
  DateTime scheduledDate;
  List<DateTime> offeredAppointments;

  VaccinationEntity();

  static VaccinationEntity fromJson(Map<String, dynamic> json) {
    return _$VaccinationEntityFromJson(json);
  }

  Map<String, dynamic> toJson() => _$VaccinationEntityToJson(this);
}
